codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/ClientReExports.kt

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.client.smithy
+
+import software.amazon.smithy.rust.codegen.core.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
+
+/** Returns a symbol for a type re-exported into crate::config
+ *  Although it is not always possible to use this, this is the preferred method for using types in config customizations
+ *  and ensures that your type will be re-exported if it is used.
+ */
+fun configReexport(type: RuntimeType): RuntimeType = RuntimeType.forInlineFun(type.name, module = ClientRustModule.config) {
```

Comment:
In an ideal world, I prefer to have all the re-exports in one place, but I understand the need to make them conditional and how that's easier with this approach. That said, I don't think we should use this for re-exports that are supposed to _always_ be present since then it could be easy to break one of these re-exports just by changing codegen in some other place.

Reply:
I see that but I think:
1. It's easier to miss items
2. We'll get a semver failure if we delete something by accident

Reply:
The other benefit is that in-situ, we're actually _using_ the rexport which I think is a benefit. We could make non-conditional re-exports be runtime types though, not opposed to that.